### ISSUE TITLE
MA0143 - False Positive is Dispose Method

### ISSUE THREAD

**Issue by sharpninja** [meziantou/Meziantou.Analyzer#689]
### Version of the Meziantou.Analyzer NuGet package

2.0.136

### Rule Identifier

MA0143

### Target Framework

net8.0

### C# Language version

C# latest

### Description

Primary Constructor parameters that need to be release when the object implements `IDisposable` should not trigger MA0143 inside the `Dispose` method pattern.

### Reproduction Steps

Minimal code:

```c#
public class A(object P) : IDisposable
{
  public void Dispose()
  {
    P = null; // triggers MA0143
  }
}
```

**Comment by meziantou** [comment]
I'm not against adding an exception for the rule with a configuration flag. However, I don't see the value of setting the parameter to `null` when you dispose the object. Most of the time this is useless because when you dispose an object graph, the initial caller also removes the "root" reference, so the GC can free the memory of the whole graph. You don't need to set all intermediate references to null.
